src/intro/hello-world.md

```diff
@@ -383,15 +388,15 @@ impl ISprite2D for Player {
         // var velocity = Vector2.UP.rotated(rotation) * speed
         // position += velocity * delta
         
-        self.sprite.rotate((self.angular_speed * delta) as f32);
+        self.base_mut().rotate((self.angular_speed * delta) as f32);
 
-        let rotation = self.sprite.get_rotation();
+        let rotation = self.base().get_rotation();
         let velocity = Vector2::UP.rotated(rotation) * self.speed as f32;
-        self.sprite.translate(velocity * delta as f32);
+        self.base_mut().translate(velocity * delta as f32);
         
         // or verbose: 
-        // self.sprite.set_position(
-        //     self.sprite.position() + velocity * delta as f32
+        // self.base_mut().set_position(
+        //     self.base().position() + velocity * delta as f32
         // );
```

Comment:
I don't think this compiles -- there is a `&mut` borrow from `base_mut()` pending when `base()` is invoked.
Maybe change it to use two separate statements, storing the position in a variable.

Reply:
I just ran into this while going through the book myself, and you're exactly right, it doesn't compile. Here's what I have, which does compile and functions correctly:

```rust
        let rotation = (self.angular_speed * delta) as f32;
        self.base_mut().rotate(rotation);

        let rotation = self.base().get_rotation();
        let velocity = Vector2::UP.rotated(rotation) * self.speed as f32;
        self.base_mut().translate(velocity * delta as f32)
```